SpinCore_pp/power_helper.py

```diff
@@ -1,8 +1,105 @@
-from numpy import *
+import numpy as np
+from numpy import r_
+import sympy as sp
+from scipy.integrate import quad
+import pyspecdata as psp
 
 "Helper functions for dealing with powers"
+# {{{ For even spacing based on phalf estimation
+def integral_as_quad(expr, lims):
+    var, a, b = lims
+    return quad(sp.lambdify(var, expr, modules="numpy"), a, b)[0]
 
 
+Emax, p, phalf, pmax = sp.symbols("E_max p p_half p_max", real=True, positive=True)
+
+
+def Ep_spacing_from_phalf(
+    est_phalf=0.2,
+    sim_Emax=30,
+    max_power=3.16,
+    aspect_ratio=1.618,
+    p_steps=20,
+    min_dBm_step=1.0,
+    three_down=True,
+    fl=None,
+):
+    """
+        Parameters
+        ==========
+        est_phalf:      float
+                        estimated power for half saturation
+        sim_Emax:       float
+                        ONLY FOR PLOTTING PURPOSES. when fl is not None, this will generate an E(p) curve using this Emax but it is not used in the actual function.
+        max_power:      float
+                        maximum power that you will send to the sample (W)
+        aspect_ratio:   float
+                        width of figure size/ height of figure size. Most likely you will not need to change this but if you want to make a different figure size aspect ratio this ensures the E(p) spacing is even within that figure
+        p_steps:        float
+                        number of power steps in the enhancement experiment
+        min_dBm_step:   float
+                        minimum stepsize that is allowed, in dBm
+        three_down:     boolean
+                        Do you want three evenly spaced powers on the way
+                        back down?
+    Return
+        ======
+        List of evenly spaced powers (dBm) for generated a nicely spaced Ep curve for a specific sample based on the estimated p_half.
+    """
+    p_array = r_[0:max_power:200j]
```

Comment:
I noticed this guy -- is this not merged into master?  Does that mean you are not using the good spacing in your experiments?